### ISSUE TITLE
`__all__.extend(...)` fails when the argument is an empty list

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#8636]
**Describe the bug**

When using `__all__.extend(...)`, if the argument is a plain `[]` or `__all__` from another module while it's empty, `reportUnsupportedDunderAll` is triggered.

By contrast, `__all__ += ...` doesn't have a same behavior.

**Code or Screenshots**

```python
__all__: list[str] = []
__all__ += []  # passes
__all__.extend([])  # fails
```

```python
# module `a`
__all__: list[str] = []

# module `b`
__all__: list[str] = []
import a
__all__ += a.__all__  # passes
__all__.extends(a.__all__)  # fails
```

**VS Code extension or command-line**

Pyright 1.1.374 CLI.

**Comment by erictraut** [comment]
This is included in pyright 1.1.375.
